tests/src/test/java/org/mozilla/javascript/tests/InterpreterFunctionPeelingNonRegressionTest.java

```diff
@@ -0,0 +1,86 @@
+package org.mozilla.javascript.tests;
+
+import static org.junit.Assert.assertEquals;
+import static org.mozilla.javascript.LambdaConstructor.CONSTRUCTOR_FUNCTION;
+
+import org.junit.Test;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.LambdaConstructor;
+import org.mozilla.javascript.LambdaFunction;
+import org.mozilla.javascript.ScriptRuntime;
+import org.mozilla.javascript.Scriptable;
+import org.mozilla.javascript.ScriptableObject;
+
+public class InterpreterFunctionPeelingNonRegressionTest {
+    @Test
+    public void testLambdaFunction() {
+        try (var cx = Context.enter()) {
+            cx.setOptimizationLevel(-1);
+            Scriptable scope = cx.initStandardObjects();
+
+            LambdaFunction makePerson =
+                    new LambdaFunction(
+                            scope,
+                            "makePerson",
+                            1,
+                            (cx1, scope1, thisObj, args) ->
+                                    new Person(ScriptRuntime.toString(args[0])));
+            scope.put("makePerson", scope, makePerson);
+
+            String source =
+                    "function testLambdaFunction() {\n"
+                            + "   return makePerson('Andrea');\n"
+                            + "}\n"
+                            + "testLambdaFunction().name";
+            Object value = cx.evaluateString(scope, source, "test", 1, null);
+            assertEquals("Andrea", value);
```

Comment:
Ok, its about the interpreter, but i think we should run as many tests as possible with all optimization levels.

Reply:
Agreed, not much of a risk to use one of the methods to "runWithAllOptimizationLevels".

Reply:
There are also some assert... methods in the Utils class, maybe you can also use one of these to make this simpler
eg. assertWithAllOptimizationLevels

Reply:
Thanks